### ISSUE TITLE
CP calculations have changed? Saved workflows are calculating differently

### ISSUE THREAD

**Issue by NemxLevram** [doxxx/ffxiv-craft-opt-web#230]
## Required Information

Class: Leatherworker

Level: 70
Craftsmanship: 1448
Control: 1352
CP: 476
Recipe Name: Slothskin Gloves of Scouting
Recipe Level: 70***
Solver Seed: 98147

Food: Tempura Platter (HQ)
Medicine: Cunning Craftsman's Tea (HQ)

## Expected Behaviour

My saved workflow for this recipe now calculates as not having enough CP for the last five steps. When I originally created the workflow about two months ago, all the calculations worked as expected. I have changed nothing in my crafter profile. Just loaded the Optimizer and the recipe and saw that it was now indicating that the recipe would not work.

Seeing as how I had changed nothing, I crafted the recipe again following the saved workflow, and everything working in-game as expected. I have no idea what changed in the Optimizer's calculations.

## Steps To Reproduce

I've attached a screenshot of the workflow, as loaded.
![saved workflow](https://user-images.githubusercontent.com/39448103/40274586-1c9aaa10-5ba8-11e8-8339-8ea6392e83cc.jpg)

**Comment by doxxx** [comment]
Does the bonus CP (67) from the food buff give the same amount in-game?

**Comment by NemxLevram** [comment]
Yes, I'm currently crafting Crimson Cider on my CUL, with the same stats as my LTW. I have 543 CP with the HQ Tempura Platter and HQ Cunning Tea.

**Comment by NemxLevram** [comment]
Perhaps it's an issue with how the Optimizer is handling the two Tricks of the Trade? I need those CP refunds for the workflow to work. I take them anywhere I can between Step 3 and 37.

The Optimizer has always handled that calculation correctly in the past. I insert / remove / shift Tricks around all the time in the workflows I create and have never had a problem before.

**Comment by doxxx** [comment]
We recently changed the way the Monte Carlo simulation is performed to simulate macro behavior more closely by default since the majority of people use it that way. In Macro Mode, unusable conditional actions are skipped instead of being repositioned to the point where the conditions are right for using them. This is probably what's causing the CP shortfall.

You can switch back to the original method by going into the Options, enabling Advanced Mode on the Monte Carlo tab and selecting Reposition under Conditional Action Handling.

**Comment by NemxLevram** [comment]
Right, I see how that advanced option is intended to work.

I updated my options as you described, gave my browser a Ctrl+F5 refresh and reloaded the workflow, but unfortunately it is still calculating the CP shortfall as -11 with the last five steps greyed-out.

**Comment by doxxx** [comment]
Could you export the sequence and paste it here so I can import it for testing?

**Comment by NemxLevram** [comment]
I think this is the export you wanted (used the "More..." button). Let me know if you needed something else.
[NemxLevram settings.zip](https://github.com/doxxx/ffxiv-craft-opt-web/files/2019930/NemxLevram.settings.zip)

**Comment by doxxx** [comment]
So the Monte Carlo results look like this now:

```
Monte Carlo Result of 500 runs
==============================
                          DUR    CP      QUA   PRG   HQ%
##     Expected Value:      0     9  27354.2 3661.1  99.8
##       Median Value:      0    11  26330.0 3676.0 100.0
##          Min Value:     -5    -7  19909.0 2834.0  63.0
##          Max Value:     15    13  63946.0 3756.0 100.0

##       Success Rate:   96.6 %
```

But the status widget and the sequence are not displaying the same state. I'll investigate.

**Comment by NemxLevram** [comment]
Ah yes, I failed to notice that.

Thanks for all your efforts on the Optimizer! It's interface is intuitive and works beautifully. It totally is the linchpin for all of my crafting activities. I always worry that it's going to go out of date at some point.

**Comment by doxxx** [comment]
Thanks, it's nice to hear that! Some parts are easy to keep up to date, like recipes because they're mostly automated, but the formulas are harder. Luckily, we have some help from people who still play the game and can gather data to adjust the formulas.
